### ISSUE TITLE
Implement interface helper does not consider what is already implemented on base classes

### ISSUE THREAD

**Issue by Tyriar** [microsoft/TypeScript#19847]
**TypeScript Version:**  2.6.1 (VS Code Insiders)

**Code**

```ts
interface MyInterface {
  foo(): void;
}

interface IBase extends MyInterface {
  bar(): void;
}

abstract class Base implements MyInterface {
  foo(): void {}
}

class MyClass extends Base implements IBase {

}

```

When I implement interface `IBase`, it's not figuring out which methods are already implemented on the base classes.

![image](https://user-images.githubusercontent.com/2193314/32576123-cf391c34-c48a-11e7-88ab-84cfbcf8b9ad.png)

**Expected behavior:**

```ts
class MyClass extends Base implements IBase {
  bar(): void {
    throw new Error("Method not implemented.");
  }
}

```

**Actual behavior:**

```ts
class MyClass extends Base implements IBase {
  bar(): void {
    throw new Error("Method not implemented.");
  }
  foo(): void {
    throw new Error("Method not implemented.");
  }

}
```

**Comment by mhegazy** [comment]
PRs welcomed...

**Comment by MhdTlb** [comment]
I would like to work on this issue but I need some guidance

**Comment by mhegazy** [comment]
sure. The code for this fix is in https://github.com/Microsoft/TypeScript/blob/master/src/services/codefixes/fixClassIncorrectlyImplementsInterface.ts

The fix should probably be in https://github.com/Microsoft/TypeScript/blob/d998e97d8c64acf1ed4b8cd508a4589c110bd3aa/src/services/codefixes/helpers.ts#L36 to change `const classMembers = classDeclaration.symbol.members;`  to ` const classMembers = checker.getPropertiesOfType(checker.getTypeAtLocation(classDeclaration));`

**Comment by shobhitchittora** [comment]
@mhegazy I've made the required changes and would like to test now. How can I test whether the issue is fixed? Also how to debug ts compiler changes?

PS: I used vscode to reproduce the issue.

**Comment by mhegazy** [comment]
VSCode ships with an older version of TypeScript. Please see [Using Newer TypeScript Versions](https://code.visualstudio.com/docs/languages/typescript#_using-newer-typescript-versions) documentation for more details on updating your VSCode to use a different version of TypeScript.

**Comment by mhegazy** [comment]
u also should write a unit test for it, see https://github.com/Microsoft/TypeScript/blob/master/tests/cases/fourslash/codeFixClassImplementInterfaceProperty.ts

**Comment by shobhitchittora** [comment]
@mhegazy @Tyriar Need help debugging ts executions from vscode. I tried putting a ```debugger;``` in my ts file hoping that it'd stop while fixing incorrectly implemented interfaces in my vscode OSSDEV build, but it does not.

Thanks in advance. 
PS: I've configured my vscode nightly build (from master) to load local ```tsserver.js```.

**Comment by Tyriar** [comment]
@mjbvz might be able to help with debugging in vscode

**Comment by shobhitchittora** [comment]
Ping @mjbvz. Can you help with this?

**Comment by mjbvz** [comment]
@shobhitchittora To debug the ts server loaded by code:

First point your vscode `typescript.tsdk` user setting your locally build version of TypeScript:

```
"typescript.tsdk": "/Users/matb/projects/TypeScript/built/local/"
```

Using a local build of code, change the `execArgv` [in this file](https://github.com/Microsoft/vscode/blob/master/extensions/typescript/src/typescriptServiceClient.ts#L325) to be `execArgv: ['--inspect=5859']`

Attach to the ts server spawned by the local build of code using:

```json
{
			"type": "node",
			"request": "attach",
			"name": "Attach to TS Server",
			"protocol": "inspector",
			"port": 5859
		}
```

**Comment by shobhitchittora** [comment]
A few updates on this thread - 

1. @mjbvz Thanks. It really helped.
2. The `TypeScript/src/services/codefixes/helpers.ts` is refactored since the last comment by @mhegazy. The method new is at - https://github.com/Microsoft/TypeScript/blob/master/src/services/codefixes/helpers.ts#L9.

Blocker- 
1. `classMembers` is coming as empty array when try to get autofix suggestion in vscode. ( refer screenshot )
2. Even the fix suggested by @mhegazy behaves the same.
<img width="1136" alt="screen shot 2017-12-20 at 3 34 12 pm" src="https://user-images.githubusercontent.com/4711855/34201797-b45e614e-e59b-11e7-83de-8dc28c9d9c2c.png">

**Comment by dragomirtitian** [comment]
This issue should be closed. It no longer happens since 3.0. (Had my eye on fixing it and found I had nothing to fix :) )

**Comment by laughinghan** [comment]
Indeed, no error at all here: https://www.typescriptlang.org/play/index.html#src=interface%20MyInterface%20%7B%0A%20%20foo()%3A%20void%3B%0A%7D%0A%0Ainterface%20IBase%20extends%20MyInterface%20%7B%0A%20%20bar()%3A%20void%3B%0A%7D%0A%0Aabstract%20class%20Base%20implements%20MyInterface%20%7B%0A%20%20foo()%3A%20void%20%7B%7D%0A%7D%0A%0Aclass%20MyClass%20extends%20Base%20implements%20IBase%20%7B%0A%20%20bar()%3A%20void%20%7B%0A%20%20%20%20throw%20new%20Error(%22Method%20not%20implemented.%22)%3B%0A%20%20%7D%0A%7D

**Comment by mjbvz** [comment]
Confirmed that this is fixed with TS 3.7.5
